lib/dolly/request.rb

```diff
@@ -40,6 +40,10 @@ def delete resource
       request :delete, full_path(resource), {}
     end
 
+    def attach resource, attachment_name, data, headers = {}
+      request :put, attachment_path(resource, attachment_name), {body: data, headers: headers}
```

Comment:
I think the body should be Base64 encoded... not sure if you should leave that to the user... maybe?